metadata_upload/metadata_upload_tutorial.Rmd

```diff
@@ -0,0 +1,623 @@
+  * make the google sheet public (to anyone with link) and publish to the web.
+  * instructions on how to fill out the template can be found under the README worksheet.
+* Database connection parameters 
+  * save in a .pgpass file in your home directory.
+  * see more information on how to create this file below in section 3.
+
+# Section 2: Setting up local instance of test database
+
+## Install Docker
+
+To run a local instance of the BETY test database, you will need to have Docker installed on your computer. 
+
+Visit [https://www.docker.com/](https://www.docker.com/) to download Docker for MAC or Windows. You will need to create a docker account.
+
+## Clone bety github repository
+
+In a directory of your choice, clone the bety repository from github:
+
+`git clone git@github.com:pecanproject/bety`
+
+Then cd to the bety directory:
+
+`cd bety`
+
+## Start up and run docker container
+
+Mount postgres container to port 5432 in docker-compose.override file:
+
+`wget https://gist.githubusercontent.com/dlebauer/a2cd7fec09280e5a01e208e91806694f/raw/7fd122055c1c7995ac2d606ec67b25402d2ae9de/docker-compose.override.yml`
+
+Start postgres:
+
+`docker-compose -p bety up -d postgres`
+
+Initialize BETY database:
+
+`docker run -ti --rm --network bety_bety -e BETY_INITIALIZE_URL='-w https://terraref.ncsa.illinois.edu/bety/dump/bety0/bety.tar.gz' pecan/bety:terra initialize`
+
+Sync with server 6 only:
+
+`docker run -ti --rm --network bety_bety -e REMOTE_SERVERS='6' pecan/bety:terra sync`
+
+Bring up full stack:
+
+`docker-compose up`
+
+# Section 3: Setting up .pgpass file
+
+A .pgpass file will be used to pass in database connection parameters to the BETY test database.
+
+In your home directory, create a `.pgpass` file and change the permissions to 600.
+
+```sh
+touch ~/.pgpass
+chmod 0600 ~/.pgpass
+```
+
+On a new line, enter the database connection parameters for the local instance of the BETY test database in the format:  
+
+`host:port:database:username:password`
+
+# Section 4: Set up for metadata upload
+
+When running this tutorial interactively:
+
+**You will need to update database connection parameters in this section on lines 117-120 to match those in your .pgpass file**
+
+**You will need to pass in the url to your google sheet in this section on line 138**
+
+## Load in packages
+
+```{r load-pack}
+# load in packages
+library(RPostgres)
+library(readxl)
+library(dplyr)
+library(googlesheets)
+```
+
+## Set options
+
+```{r set-option}
+options(scipen = 999)
+```
+
+## Connect to database; Provide connection parameters here (REQUIRED)
+
+_Please make sure the following database connection parameters (lines 117-120) match those that you have inputted in your .pgpass file. Make any needed changes._
+
+_The following connection parameters will be used when running this Rmd file from command line_
+
+```{r create-dbcon}
+# create database connection using parameters in .pgpass file
+dbcon <- dbConnect(RPostgres::Postgres(),
```

Comment:
something like read in from .pgpass, separate by ':', then use the values to set the connection

```r
pgpass <- readLines('.pgpass') 
 # https://tidyr.tidyverse.org/reference/separate.html
con <- as.list(separate(pgpass))
```